ext/src/dll/CMakeLists.txt

```diff
@@ -6,6 +6,10 @@ set(OPENTELEMETRY_EXPORT_DEF
 
 add_library(opentelemetry_cpp SHARED dllmain.cc ${OPENTELEMETRY_EXPORT_DEF})
 
+if(OPENTELEMETRY_STL_VERSION)
+  add_compile_definitions(OPENTELEMETRY_STL_VERSION)
```

Comment:
Just wondering, should this be added into opentelemetry_api? Or should we use `#if defined(OPENTELEMETRY_STL_VERSION) && OPENTELEMETRY_STL_VERSION >= 2017` below?